src/game_state/player.rs

```diff
@@ -87,52 +134,56 @@ pub fn start_player_state(
                 broadcast_packet!(
                     messenger,
                     Packet::EntityLookAndMove(EntityLookAndMove {
-                        entity_id: player_clone.conn_id.as_u128() as i32,
+                        entity_id: player.entity_id,
                         delta_x: position_delta.x,
                         delta_y: position_delta.y,
                         delta_z: position_delta.z,
                         yaw: 0,
                         pitch: 0,
                         on_ground: false,
-                    })
+                    }),
+                    Some(player.conn_id),
+                    true
                 )
                 .unwrap();
                 player_clone.position = msg.new_position;
                 players.insert(msg.conn_id, player_clone);
             }
             PlayerStateOperations::Report(msg) => {
-                players.values().for_each(|player| {
-                    let player_clone = player.clone();
-                    send_packet!(
-                        messenger,
-                        msg.conn_id,
-                        Packet::PlayerInfo(PlayerInfo {
-                            action: 0,
-                            number_of_players: 1, //send each player in an individual packet for now
-                            uuid: player_clone.uuid.as_u128(),
-                            name: player_clone.name.clone(),
-                            number_of_properties: 0,
-                            gamemode: 1,
-                            ping: 100,
-                            has_display_name: false,
-                        })
-                    )
-                    .unwrap();
-                    send_packet!(
-                        messenger,
-                        msg.conn_id,
-                        Packet::SpawnPlayer(SpawnPlayer {
-                            entity_id: player.entity_id,
-                            uuid: player_clone.uuid.as_u128(),
-                            x: player_clone.position.x,
-                            y: player_clone.position.y,
-                            z: player_clone.position.z,
-                            yaw: 0,
-                            pitch: 0,
-                            entity_metadata_terminator: 0xff,
-                        })
-                    )
-                    .unwrap();
+                players.iter().for_each(|(conn_id, player)| {
+                    if *conn_id != msg.conn_id {
+                        let player_clone = player.clone();
+                        send_packet!(
```

Comment:
This one is fun. When we send an "entity moved packet" to the player who caused it to happen, it causes an infinite loop where they move in response, then another move packet is sent, etc. Because of this interaction we need to introduce the source_conn_id parameter that indicates what player caused this event and be sure they don't hear about the result